qa/coccinelle/malloc-error-check.cocci

```diff
@@ -58,6 +58,8 @@ FAIL_IF(x == NULL)
 FAIL_IF(unlikely(x == NULL))
 |
 FAIL_IF_NULL(x)
+|
+return x;
```

Comment:
Cocci cannot however detect if the caller does indeed check that value, so this could miss some things?

Reply:
Anyways, I think its this fix to cocci or make cocci happy with a superfluous return check which may trigger other linters: https://github.com/OISF/suricata/commit/d3cc5d89df346649d259d47e4447fa4ae86f9bf0

Reply:
I think we should fix CI tooling because the code is good.

Reply:
Can't tell if this is a fix, or just makes cocci more permissive.

Reply:
> Can't tell if this is a fix, or just makes cocci more permissive.

It is making it more permissive.  Cocci, as far as I can tell does not have the ability to jump to callers of this function and validate the return check there.  So wrappers like this need superfluous checks, or this more permissive Cocci check.  At least in my observation so far.